ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -49,8 +49,7 @@
   private final HashMap<String, RequestMetadata> connectionIdToRequestInFlight;
   private final AtomicLong numPendingRequests;
   private final int checkoutTimeoutMs;
-  // @todo: this needs to be empirically determined.
-  private final int POLL_TIMEOUT_MS = 1;
+  private final int POLL_TIMEOUT_MS = 300;
```

Comment:
This should not be configurable as the user is only concerned with request timeouts. But yes, this should be a function of the request timeout and should be smaller. Will make the `sendAndPoll()` take an explicit timeout and will pass in a value that is an order of magnitude smaller than the request timeout.